### PR TITLE
Add clang-tools to required tools for ci_static_analysis_clang

### PR DESCRIPTION
**Pull request by nlohmann**
Fix build, see https://github.com/nlohmann/json/pull/3723#issuecomment-1236280002

### CODE REVIEW

**Comment by falbrechtskirchinger** [comment]
You can cancel my workflows since they're going to fail anyway.

File: tests/src/unit-32bit.cpp [hunk 1]
```diff
@@ -34,9 +34,9 @@ TEST_CASE_TEMPLATE_DEFINE("value_in_range_of trait", T, value_in_range_of_test)
     constexpr bool max_in_range = T::max_in_range;
 
     type val_min = std::numeric_limits<type>::min();
-    type val_min2 = val_min + 1;
+    type const val_min2 = val_min + 1;
     type val_max = std::numeric_limits<type>::max();
-    type val_max2 = val_max - 1;
+    type const val_max2 = val_max - 1;
```
Comment (falbrechtskirchinger) [inline tests/src/unit-32bit.cpp:39]: The other two can be const too.

File: tests/src/unit-capacity.cpp [hunk 2]
```diff
@@ -416,7 +416,7 @@ TEST_CASE("capacity")
     {
         SECTION("boolean")
         {
-            json j = true;
+            json const j = true;
             const json j_const = true;
```
Comment (falbrechtskirchinger) [inline tests/src/unit-capacity.cpp:420]: This is one of those cases I mentioned on Discord. This should use `NOLINT` instead.

Comment (falbrechtskirchinger) [inline tests/src/unit-capacity.cpp]: This is one of those cases I mentioned on Discord. This should use `NOLINT` instead.
In fact, this applies to all occurrences in this file AFAICT.

File: tests/src/unit-32bit.cpp (hunk 1 above)
Comment (falbrechtskirchinger) [inline tests/src/unit-32bit.cpp]: The other two can be const too.

File: tests/src/unit-class_const_iterator.cpp [hunk 3]
```diff
@@ -20,27 +20,27 @@ TEST_CASE("const_iterator class")
         {
             SECTION("null")
             {
-                json j(json::value_t::null);
-                json::const_iterator it(&j);
+                json const j(json::value_t::null);
+                json::const_iterator const it(&j);
```
Comment (falbrechtskirchinger) [inline tests/src/unit-class_const_iterator.cpp:24]: This seems fine but is another one of those corner cases we should pay attention to in this PR.

File: tests/src/unit-constructor1.cpp [hunk 4]
```diff
@@ -1311,7 +1311,7 @@ TEST_CASE("constructors")
                     CHECK(j_new == json::object());
                 }
                 {
-                    json jobject = {{"a", "a"}, {"b", 1}, {"c", 17u}};
+                    json const jobject = {{"a", "a"}, {"b", 1}, {"c", 17u}};
                     json j_new(jobject.cbegin(), jobject.cbegin());
                     CHECK(j_new == json::object());
```
Comment (falbrechtskirchinger) [inline tests/src/unit-constructor1.cpp]: `j_new` could/should be `const` too? Applies several more times in this file.

File: tests/src/unit-ubjson.cpp [hunk 5]
```diff
@@ -1703,7 +1703,7 @@ TEST_CASE("UBJSON")
             SECTION("optimized version (length only)")
             {
                 // create vector with two elements of the same type
-                std::vector<uint8_t> v_TU = {'[', '#', 'U', 2, 'T', 'T'};
+                std::vector<uint8_t> const v_TU = {'[', '#', 'U', 2, 'T', 'T'};
                 std::vector<uint8_t> v_T = {'[', '#', 'i', 2, 'T', 'T'};
                 std::vector<uint8_t> v_F = {'[', '#', 'i', 2, 'F', 'F'};
                 std::vector<uint8_t> v_Z = {'[', '#', 'i', 2, 'Z', 'Z'};
```
Comment (falbrechtskirchinger) [inline tests/src/unit-ubjson.cpp]: Pretty much all of these vectors can and should be `const`. Applies in many more places and I probably scrolled past them in the other binary format tests.

File: tests/src/unit-bjdata.cpp [hunk 6]
```diff
@@ -119,9 +119,9 @@ TEST_CASE_TEMPLATE_DEFINE("value_in_range_of trait", T, value_in_range_of_test)
     constexpr bool max_in_range = T::max_in_range;
 
     type val_min = std::numeric_limits<type>::min();
-    type val_min2 = val_min + 1;
+    type const val_min2 = val_min + 1;
     type val_max = std::numeric_limits<type>::max();
-    type val_max2 = val_max - 1;
+    type const val_max2 = val_max - 1;
```
Comment (falbrechtskirchinger) [inline tests/src/unit-bjdata.cpp]: These can all be `const`.

File: tests/src/unit-bjdata.cpp [hunk 7]
```diff
@@ -2233,13 +2233,13 @@ TEST_CASE("BJData")
         {
             // create a single-character string for all number types
             std::vector<uint8_t> s_i = {'S', 'i', 1, 'a'};
```
Comment (falbrechtskirchinger) [inline tests/src/unit-bjdata.cpp:2271]: Can be `const`.

File: tests/src/unit-bjdata.cpp [hunk 8]
```diff
@@ -2284,7 +2284,7 @@ TEST_CASE("BJData")
             SECTION("optimized version (length only)")
             {
                 // create vector with two elements of the same type
-                std::vector<uint8_t> v_TU = {'[', '#', 'U', 2, 'T', 'T'};
+                std::vector<uint8_t> const v_TU = {'[', '#', 'U', 2, 'T', 'T'};
                 std::vector<uint8_t> v_T = {'[', '#', 'i', 2, 'T', 'T'};
                 std::vector<uint8_t> v_F = {'[', '#', 'i', 2, 'F', 'F'};
                 std::vector<uint8_t> v_Z = {'[', '#', 'i', 2, 'Z', 'Z'};
```
Comment (falbrechtskirchinger) [inline tests/src/unit-bjdata.cpp]: Can all be `const`? Occurs many more times in this file.

**Review by falbrechtskirchinger — CHANGES_REQUESTED** [review]
I expected more cases where `NOLINT` might be needed.

I don't understand why variables that curiously always seem to appear in `CHECK(var == ...)` and can clearly be `const` are not being complained about.

File: tests/src/unit-capacity.cpp (hunk 2 above)
Comment (falbrechtskirchinger) [reply tests/src/unit-capacity.cpp]: Without `NOLINT`:
```cpp
    json j = true;
    const json j_const(j);
```
Fixed in #3738.

File: tests/src/unit-32bit.cpp (hunk 1 above)
Comment (falbrechtskirchinger) [reply tests/src/unit-32bit.cpp]: Fixed in #3738.

File: tests/src/unit-constructor1.cpp (hunk 4 above)
Comment (falbrechtskirchinger) [reply tests/src/unit-constructor1.cpp]: Fixed in #3738.

File: tests/src/unit-ubjson.cpp (hunk 5 above)
Comment (falbrechtskirchinger) [reply tests/src/unit-ubjson.cpp]: Fixed in #3738.

File: tests/src/unit-bjdata.cpp (hunk 6 above)
Comment (falbrechtskirchinger) [reply tests/src/unit-bjdata.cpp]: Fixed in #3738.

File: tests/src/unit-bjdata.cpp (hunk 7 above)
Comment (falbrechtskirchinger) [reply tests/src/unit-bjdata.cpp:2271]: Fixed in #3738.

File: tests/src/unit-bjdata.cpp (hunk 8 above)
Comment (falbrechtskirchinger) [reply tests/src/unit-bjdata.cpp]: Fixed in #3738.

**Comment by falbrechtskirchinger** [comment]
Discord.

**Review by falbrechtskirchinger — APPROVED** [review]
Looks good to me.
